src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtils.java

```diff
@@ -173,66 +176,61 @@ public static double parseDouble(String text, int type) {
                     radix = BASE_16;
                     txt = txt.substring(2);
                 }
-                else if (txt.charAt(0) == '0') {
+                else if (txt.startsWith("0b") || txt.startsWith("0B")) {
+                    radix = BASE_2;
+                    txt = txt.substring(2);
+                }
+                else if (CommonUtils.startsWithChar(txt, '0')) {
                     radix = BASE_8;
                     txt = txt.substring(1);
                 }
-                if (CommonUtils.endsWithChar(txt, 'L') || CommonUtils.endsWithChar(txt, 'l')) {
-                    txt = txt.substring(0, txt.length() - 1);
-                }
-                if (!txt.isEmpty()) {
-                    if (type == TokenTypes.NUM_INT) {
-                        result = parseInt(txt, radix);
-                    }
-                    else {
-                        result = parseLong(txt, radix);
-                    }
-                }
+                result = parseNumber(txt, radix, type);
                 break;
             default:
+                result = Double.NaN;
                 break;
         }
         return result;
     }
 
     /**
-     * Parses the string argument as a signed integer in the radix specified by
+     * Parses the string argument as an integer or a long in the radix specified by
      * the second argument. The characters in the string must all be digits of
-     * the specified radix. Handles negative values, which method
-     * java.lang.Integer.parseInt(String, int) does not.
+     * the specified radix.
      * @param text the String containing the integer representation to be
      *     parsed. Precondition: text contains a parsable int.
      * @param radix the radix to be used while parsing text.
-     * @return the integer represented by the string argument in the specified radix.
+     * @param type the token type of the text. Should be a constant of
+     * {@link TokenTypes}.
```

Comment:
Please indent this line by +4.

@romani I'm surprised JavadocTagContinuationIndentation didn't flag this.

Reply:
done